core/src/main/scala/org/broadinstitute/dsde/rawls/metrics/InstrumentationDirectives.scala

```diff
@@ -2,11 +2,27 @@ package org.broadinstitute.dsde.rawls.metrics
 
 import java.util.concurrent.TimeUnit
 
-import spray.routing.Directive0
+import shapeless.HNil
+import spray.http.Uri
+import spray.routing._
 import spray.routing.directives.BasicDirectives.mapHttpResponse
 import spray.routing.directives.MiscDirectives.requestInstance
+import spray.routing.directives.PathDirectives._
 
 trait InstrumentationDirectives extends RawlsInstrumented {
+
+  // Like Segment in that it matches and consumes any path segment, but does not extract a value.
+  private val SegmentIgnore: PathMatcher0 = Segment.hmap(_ => HNil)
+
+  private val redactWorkflowIds: PathMatcher1[String] =
+    (Slash ~ "workspaces") / SegmentIgnore / SegmentIgnore / "submissions" / SegmentIgnore / "workflows" / (Segment ~ SegmentIgnore.repeat(separator = Slash))
```

Comment:
Agreed! Originally I did this with regexes but then decided to use the spray pathmatcher instead. I still feel it's a little nicer than regexes in this case.